src/Functions/FunctionsStringDistance.cpp

```diff
@@ -518,32 +518,206 @@ using FunctionJaroWinklerSimilarity = FunctionsStringSimilarity<FunctionStringDi
+    };
+    FunctionDocumentation::IntroducedIn introduced_in_jaccard_utf8 = {23, 11};
+    FunctionDocumentation documentation_jaccard_utf8 = {description_jaccard_utf8, syntax_jaccard_utf8, arguments_jaccard_utf8, returned_value_jaccard_utf8, examples_jaccard_utf8, introduced_in_jaccard_utf8, category};
+
+    FunctionDocumentation::Description description_jaro = R"(
+Calculates the [Jaro similarity](https://en.wikipedia.org/wiki/Jaro%E2%80%93Winkler_distance#Jaro_similarity) between two byte strings.
+)";
+    FunctionDocumentation::Syntax syntax_jaro = "jaroSimilarity(s1, s2)";
+    FunctionDocumentation::Arguments arguments_jaro = {
+        {"s1", "First input string.", {"String"}},
+        {"s2", "Second input string.", {"String"}}
+    };
+    FunctionDocumentation::ReturnedValue returned_value_jaro = {"Returns the Jaro similarity between the two strings.", {"Float64"}};
+    FunctionDocumentation::Examples examples_jaro = {
+    {
+        "Usage example",
+        "SELECT jaroSimilarity('clickhouse', 'click')",
+        R"(
+┌─jaroSimilarity('clickhouse', 'click')─┐
+│                    0.8333333333333333 │
+└───────────────────────────────────────┘
+        )"
+    }
+    };
+    FunctionDocumentation::IntroducedIn introduced_in_jaro = {24, 1};
+    FunctionDocumentation documentation_jaro = {description_jaro, syntax_jaro, arguments_jaro, returned_value_jaro, examples_jaro, introduced_in_jaro, category};
+
+    FunctionDocumentation::Description description_jaro_winkler = R"(
+Calculates the [Jaro-Winkler similarity](https://en.wikipedia.org/wiki/Jaro%E2%80%93Winkler_distance) between two byte strings.
+)";
+    FunctionDocumentation::Syntax syntax_jaro_winkler = "jaroWinklerSimilarity(s1, s2)";
+    FunctionDocumentation::Arguments arguments_jaro_winkler = {
+        {"s1", "First input string.", {"String"}},
+        {"s2", "Second input string.", {"String"}}
+    };
+    FunctionDocumentation::ReturnedValue returned_value_jaro_winkler = {"Returns the Jaro-Winkler similarity between the two strings.", {"Float64"}};
+    FunctionDocumentation::Examples examples_jaro_winkler = {
+    {
+        "Usage example",
+        "SELECT jaroWinklerSimilarity('clickhouse', 'click')",
+        R"(
+┌─jaroWinklerSimilarity('clickhouse', 'click')─┐
+│                           0.8999999999999999 │
+└──────────────────────────────────────────────┘
+        )"
+    }
+    };
+    FunctionDocumentation::IntroducedIn introduced_in_jaro_winkler = {24, 1};
+    FunctionDocumentation documentation_jaro_winkler = {description_jaro_winkler, syntax_jaro_winkler, arguments_jaro_winkler, returned_value_jaro_winkler, examples_jaro_winkler, introduced_in_jaro_winkler, category};
 
-    factory.registerFunction<FunctionByteHammingDistance>(
-        FunctionDocumentation{.description = R"(Calculates Hamming distance between two byte-strings.)", .category = category_string});
+    factory.registerFunction<FunctionByteHammingDistance>(documentation_hamming);
```

Comment:
(The [SMASH  PR](https://github.com/ClickHouse/ClickHouse/pull/79461) is unfortunately still open, otherwise we'd have a super cool string distance here)